sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java

```diff
@@ -788,23 +986,33 @@ private void recordAddressResolutionStatistics(
                     addressResolutionStatistics.getStartTimeUTC(),
                     addressResolutionStatistics.getEndTimeUTC());
 
-                Timer addressResolutionLatencyMeter = Timer
-                    .builder(nameOf("rntbd.addressResolution.latency"))
-                    .description("Address resolution latency")
-                    .maximumExpectedValue(Duration.ofSeconds(6))
-                    .publishPercentiles(0.95, 0.99)
-                    .publishPercentileHistogram(true)
-                    .tags(addressResolutionTags)
-                    .register(compositeRegistry);
-                addressResolutionLatencyMeter.record(latency);
+                CosmosMeterOptions latencyOptions = clientAccessor.getMeterOptions(
+                    client,
+                    CosmosMeterName.DIRECT_ADDRESS_RESOLUTION_LATENCY);
+                if (latencyOptions.isEnabled()) {
+                    Timer addressResolutionLatencyMeter = Timer
+                        .builder(latencyOptions.getMeterName().toString())
+                        .description("Address resolution latency")
+                        .maximumExpectedValue(Duration.ofSeconds(6))
+                        .publishPercentiles(optionsAccessor.getPercentiles(latencyOptions))
+                        .publishPercentileHistogram(optionsAccessor.isHistogramPublishingEnabled(latencyOptions))
+                        .tags(getEffectiveTags(addressResolutionTags, latencyOptions))
+                        .register(compositeRegistry);
+                    addressResolutionLatencyMeter.record(latency);
```

Comment:
what would happen if the latency is higher than 6s? will it be counted in the 6s bucket?

Reply:
Yes